### PR TITLE
msc_tenant: Manage tenants on ACI MultiSite

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This includes the new msc_tenant module and integration tests.

This PR requires #46116 to be merged.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
msc_tenant

##### ANSIBLE VERSION
v2.8

### CODE REVIEW

**Comment by dagwieers** [comment]
cc @kbreit

File: lib/ansible/modules/network/aci/msc_tenant.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    elif state == 'present':
+        msc.previous = msc.existing
+
+        msc.sanitize(dict(
+            description=description,
+            id=tenant_id,
+            name=tenant,
+            displayName=display_name,
+            siteAssociations=[],
+            userAssociations=[dict(userId="0000ffff0000000000000020")],
```
Comment (kbreit) [inline lib/ansible/modules/network/aci/msc_tenant.py:163]: Is the userId static or should it update?

File: lib/ansible/modules/network/aci/msc_tenant.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        if msc.existing:
+            if not issubset(msc.sent, msc.existing):
+                if module.check_mode:
+                    msc.existing = msc.proposed
+                else:
+                    msc.existing = msc.request(path, method='PUT', data=msc.sent)
+        else:
+            if module.check_mode:
+                msc.existing = msc.proposed
+            else:
```
Comment (kbreit) [inline lib/ansible/modules/network/aci/msc_tenant.py:175]: msc.changed?

File: lib/ansible/modules/network/aci/msc_tenant.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/network/aci/msc_tenant.py:163]: It is a placeholder at this time, the lookup function has not been made yet. Same for siteAssociations.

File: lib/ansible/modules/network/aci/msc_tenant.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/network/aci/msc_tenant.py:175]: Is taken care of by `fail_json()` and `exist_json()`.

**Pull request merged**